marconi-core/src/Marconi/Core/Experiment/Class.hs

```diff
@@ -211,14 +212,19 @@ class AppendResult m event query indexer where
 class Closeable m indexer where
   close :: indexer event -> m ()
 
--- | We know how far an indexer went in the indexation of events
-class IsSync m event indexer where
-  -- | Last sync of the indexer
+-- | We know how far an indexer went in the indexation of events.
+class (Functor m) => IsSync m event indexer where
+  -- | Last sync point of the indexer.
   lastSyncPoint :: indexer event -> m (Point event)
 
+  -- | Last sync points of the indexer. By default, wrap the point from 'lastSyncPoint' into a list.
+  -- If you need more than one element, then you need to reimplement this function.
+  lastSyncPoints :: Word -> indexer event -> m [Point event]
+  lastSyncPoints _ indexer = List.singleton <$> lastSyncPoint indexer
```

Comment:
Maybe remove it and require every indexer to implement it?

Reply:
You mean remove the default implementation? If so, yes, as it's easy enough to implement.